### ISSUE TITLE
Bug: Minimum spanning tree ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by tomasonjo** [neo4j/graph-data-science#8]
Using the transport graph:

```
// Import nodes
WITH "https://github.com/neo4j-graph-analytics/book/raw/master/data/transport-nodes.csv"
AS uri
LOAD CSV WITH HEADERS FROM uri  AS row
MERGE (place:Place {id:row.id});
// Import relationships
WITH "https://github.com/neo4j-graph-analytics/book/raw/master/data/transport-relationships.csv"
AS uri
LOAD CSV WITH HEADERS FROM uri AS row
MATCH (origin:Place {id: row.src})
MATCH (destination:Place {id: row.dst})
MERGE (origin)-[:EROAD {distance: toInteger(row.cost)}]->(destination);
```

Loading the graph in memory:

```
CALL gds.graph.create('spanningtree_example', 'Place', {EROAD:{type:'EROAD',projection:'Undirected',properties:'distance'}});
```

And finally running the algorithm:

```
MATCH (n:Place {id:"Amsterdam"})
CALL gds.alpha.spanningTree.minimum.write('spanningtree_example', {weightWriteProperty:'cost', startNodeId: id(n), writeProperty:'MNIST', relationshipWeightProperty:'distance'})
YIELD createMillis, computeMillis, writeMillis, effectiveNodeCount
RETURN createMillis, computeMillis, writeMillis, effectiveNodeCount
```

I get an error:

> Failed to invoke procedure `gds.alpha.spanningTree.minimum.write`: Caused by: java.lang.ArrayIndexOutOfBoundsException: 30258

**Comment by tomasonjo** [comment]
Here is the stacktrace in logs

> neo4j-gds     | 2020-02-20 07:29:00.880+0000 WARN  Computation failed 30258
> neo4j-gds     | java.lang.ArrayIndexOutOfBoundsException: 30258
> neo4j-gds     | 	at org.neo4j.graphalgo.core.huge.AdjacencyOffsets$SinglePageOffsets.get(AdjacencyOffsets.java:126)
> neo4j-gds     | 	at org.neo4j.graphalgo.core.huge.HugeGraph.adjacencyCursorForIteration(HugeGraph.java:399)
> neo4j-gds     | 	at org.neo4j.graphalgo.core.huge.HugeGraph.runForEach(HugeGraph.java:389)
> neo4j-gds     | 	at org.neo4j.graphalgo.core.huge.HugeGraph.forEachRelationship(HugeGraph.java:304)
> neo4j-gds     | 	at org.neo4j.graphalgo.impl.spanningTrees.Prim.compute(Prim.java:88)
> neo4j-gds     | 	at org.neo4j.graphalgo.impl.spanningTrees.Prim.compute(Prim.java:49)
> neo4j-gds     | 	at org.neo4j.graphalgo.AlgoBaseProc.lambda$compute$1(AlgoBaseProc.java:296)
> neo4j-gds     | 	at org.neo4j.graphalgo.BaseProc.runWithExceptionLogging(BaseProc.java:84)
> neo4j-gds     | 	at org.neo4j.graphalgo.AlgoBaseProc.compute(AlgoBaseProc.java:292)
> neo4j-gds     | 	at org.neo4j.graphalgo.AlgoBaseProc.compute(AlgoBaseProc.java:257)
> neo4j-gds     | 	at org.neo4j.graphalgo.spanningtree.SpanningTreeProc.computeAndWrite(SpanningTreeProc.java:76)
> neo4j-gds     | 	at org.neo4j.graphalgo.spanningtree.SpanningTreeProc.minimumSpanningTree(SpanningTreeProc.java:63)
> neo4j-gds     | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> neo4j-gds     | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> neo4j-gds     | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> neo4j-gds     | 	at java.lang.reflect.Method.invoke(Method.java:498)
> neo4j-gds     | 	at org.neo4j.kernel.impl.proc.ReflectiveProcedureCompiler$ReflectiveProcedure.apply(ReflectiveProcedureCompiler.java:664)
> neo4j-gds     | 	at org.neo4j.kernel.impl.proc.ProcedureRegistry.callProcedure(ProcedureRegistry.java:219)
> neo4j-gds     | 	at org.neo4j.kernel.impl.proc.Procedures.callProcedure(Procedures.java:290)
> neo4j-gds     | 	at org.neo4j.kernel.impl.newapi.AllStoreHolder.callProcedure(AllStoreHolder.java:1098)
> neo4j-gds     | 	at org.neo4j.kernel.impl.newapi.AllStoreHolder.procedureCallWrite(AllStoreHolder.java:884)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$$anonfun$4.apply(CallSupport.scala:73)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$$anonfun$4.apply(CallSupport.scala:73)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callProcedure(CallSupport.scala:167)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callReadWriteProcedure(CallSupport.scala:74)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundQueryContext.callReadWriteProcedure(TransactionBoundQueryContext.scala:977)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$callReadWriteProcedure$1.apply(ExceptionTranslatingQueryContext.scala:169)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext$$anonfun$callReadWriteProcedure$1.apply(ExceptionTranslatingQueryContext.scala:169)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$class.translateException(ExceptionTranslationSupport.scala:33)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.translateException(ExceptionTranslatingQueryContext.scala:42)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslationSupport$class.translateIterator(ExceptionTranslationSupport.scala:47)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.translateIterator(ExceptionTranslatingQueryContext.scala:42)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.ExceptionTranslatingQueryContext.callReadWriteProcedure(ExceptionTranslatingQueryContext.scala:169)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.DelegatingQueryContext.callReadWriteProcedure(DelegatingQueryContext.scala:253)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.EagerReadWriteCallMode.callProcedure(ProcedureCallMode.scala:66)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe.org$neo4j$cypher$internal$runtime$interpreted$pipes$ProcedureCallPipe$$call(ProcedureCallPipe.scala:90)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe$$anon$$$$e6882d5096c8b8441ac36d3c0cc459e$$$$ateResultsByAppending$1.apply(ProcedureCallPipe.scala:73)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe$$anon$$$$e6882d5096c8b8441ac36d3c0cc459e$$$$ateResultsByAppending$1.apply(ProcedureCallPipe.scala:71)
> neo4j-gds     | 	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:435)
> neo4j-gds     | 	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:441)
> neo4j-gds     | 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
> neo4j-gds     | 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:409)
> neo4j-gds     | 	at org.neo4j.cypher.internal.runtime.RuntimeJavaValueConverter$feedIteratorToVisitable.accept(RuntimeJavaValueConverter.scala:53)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.runtime.PipeExecutionResult.accept(PipeExecutionResult.scala:76)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.StandardInternalExecutionResult.materializeResult(StandardInternalExecutionResult.scala:77)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.v3_5.runtime.executionplan.StandardInternalExecutionResult.initiate(StandardInternalExecutionResult.scala:59)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.ClosingExecutionResult$$anonfun$initiate$1.apply$mcV$sp(ClosingExecutionResult.scala:65)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.ClosingExecutionResult$$anonfun$initiate$1.apply(ClosingExecutionResult.scala:65)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.ClosingExecutionResult$$anonfun$initiate$1.apply(ClosingExecutionResult.scala:65)
> neo4j-gds     | 	at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.ClosingExecutionResult.org$neo4j$cypher$internal$compatibility$ClosingExecutionResult$$safely(ClosingExecutionResult.scala:167)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.ClosingExecutionResult.initiate(ClosingExecutionResult.scala:65)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.ClosingExecutionResult$.wrapAndInitiate(ClosingExecutionResult.scala:214)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:216)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery$$anonfun$execute$3.apply(CypherCurrentCompiler.scala:190)
> neo4j-gds     | 	at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
> neo4j-gds     | 	at org.neo4j.cypher.internal.compatibility.CypherCurrentCompiler$CypherExecutableQuery.execute(CypherCurrentCompiler.scala:223)
> neo4j-gds     | 	at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:101)
> neo4j-gds     | 	at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:85)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.TransactionStateMachineV1SPI$BoltResultHandleV1.start(TransactionStateMachineV1SPI.java:189)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:506)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:300)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:269)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:110)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:98)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.ReadyState.processRunMessage(ReadyState.java:140)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.ReadyState.processRunMessage(ReadyState.java:100)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.ReadyState.process(ReadyState.java:61)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.nextState(BoltStateMachineV1.java:144)
> neo4j-gds     | 	at org.neo4j.bolt.v1.runtime.BoltStateMachineV1.process(BoltStateMachineV1.java:92)
> neo4j-gds     | 	at org.neo4j.bolt.messaging.BoltRequestMessageReader.lambda$doRead$1(BoltRequestMessageReader.java:89)
> neo4j-gds     | 	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:191)
> neo4j-gds     | 	at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:139)
> neo4j-gds     | 	at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:171)
> neo4j-gds     | 	at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:154)
> neo4j-gds     | 	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604)
> neo4j-gds     | 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> neo4j-gds     | 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> neo4j-gds     | 	at java.lang.Thread.run(Thread.java:748)

**Comment by breakanalysis** [comment]
We have identified and addressed the bug caused the reported failure. The fix will be out in GDS 0.9.2.
